tests/unit_tests/test_kmers.py

```diff
@@ -0,0 +1,260 @@
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+Count, normalize and embed k-mers given nucleotide sequences
+"""
+
+import pytest
+import os
+
+import pandas as pd
+from unittest.mock import patch, MagicMock
+
+from autometa.common import kmers
+from autometa.common.exceptions import TableFormatError
+
+assembly = os.path.join("tests", "data", "metagenome.fna")
+test_df = pd.DataFrame(
+    {
+        "contig": [
+            "NODE_1_length_1389215_cov_225.275",
+            "NODE_2_length_1166739_cov_224.155",
+            "NODE_3_length_1063064_cov_225.095",
+            "NODE_4_length_1031470_cov_223.812",
+            "NODE_5_length_937195_cov_225.122",
+        ],
+        "AAAAA": [170, 1688, 143, 347, 156],
+        "AAAAG": [412, 1354, 345, 331, 317],
+        "AAATG": [536, 1513, 286, 325, 378],
+    }
+)
+test_df.set_index("contig", inplace=True)
+
+
+def test_revcomp():
+    string = "AAATTGCGCCCCCG"
+    rev_string = kmers._revcomp(string)
+    assert rev_string == "CGGGGGCGCAATTT"
+    assert type(rev_string) is str
+    assert len(rev_string) == 14
+
+
+def test_init_kmers():
+    unique_kmers_5 = kmers.init_kmers(kmer_size=5)
+    assert type(unique_kmers_5) is dict
+    assert len(unique_kmers_5) == 512
+    unique_kmers_3 = kmers.init_kmers(kmer_size=3.0)
+    assert type(unique_kmers_3) is dict
+    assert len(unique_kmers_3) == 32
+    unique_kmers_5 = kmers.init_kmers(kmer_size="5")
+    assert type(unique_kmers_5) is dict
+    assert len(unique_kmers_5) == 512
+    unique_kmers_3 = kmers.init_kmers(kmer_size="3.0")
+    assert type(unique_kmers_3) is dict
+    assert len(unique_kmers_3) == 32
+    with pytest.raises(TypeError):
+        kmers.init_kmers(kmer_size=5.5)
+    with pytest.raises(TypeError):
+        kmers.init_kmers(kmer_size="5.5")
+
+
+@patch("os.path.getsize", return_value=2 * 1024 * 1024)
+def test_load(patched_file_size):
+    kmer_table_fpath = MagicMock(
+        return_value="path_to_kmer_frequency_table",
+        name="path to input kmer frequency table",
+    )
+    with pytest.raises(FileNotFoundError):
+        kmers.load("kmer_fpath")
+    with pytest.raises(TableFormatError):
+        kmers.load(kmer_table_fpath)
+    assert patched_file_size.called is True
+
+
+@patch("autometa.common.kmers.seq_counter")
+@patch("autometa.common.kmers.mp_counter")
+def test_count(pacthed_mp_counter, pacthed_seq_counter):
+    with pytest.raises(TypeError):
+        kmers.count(assembly=assembly, size=5.5)
+    with pytest.raises(TypeError):
+        kmers.count(assembly=assembly, size="5.5")
+
+    out_df = kmers.count(assembly=assembly, size=5.0)
```

Comment:
`size=5.0` should result in an error.

We might as well decrease the computational complexity of the counting as well

```suggestion
    out_df = kmers.count(assembly=assembly, size=3)
```